src/sources/kafka.rs

```diff
@@ -0,0 +1,331 @@
+use crate::{
+    event::{self, Event},
+    topology::config::{DataType, GlobalOptions, SourceConfig},
+};
+use bytes::Bytes;
+use futures::{future, sync::mpsc, Future, Poll, Sink, Stream};
+use owning_ref::OwningHandle;
+use rdkafka::{
+    config::ClientConfig,
+    consumer::{Consumer, DefaultConsumerContext, MessageStream, StreamConsumer},
+    error::KafkaResult,
+    message::{BorrowedMessage, Message},
+};
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct KafkaSourceConfig {
+    bootstrap_servers: String,
+    topics: Vec<String>,
+    group_id: String,
+    #[serde(default = "default_auto_offset_reset")]
+    auto_offset_reset: String,
+    #[serde(default = "default_session_timeout_ms")]
+    session_timeout_ms: u64,
+    host_key: Option<String>,
+    key_field: Option<String>,
+}
+
+fn default_session_timeout_ms() -> u64 {
+    10000 // default in librdkafka
+}
+
+fn default_auto_offset_reset() -> String {
+    "largest".into() // default in librdkafka
+}
+
+#[typetag::serde(name = "kafka")]
+impl SourceConfig for KafkaSourceConfig {
+    fn build(
+        &self,
+        _name: &str,
+        _globals: &GlobalOptions,
+        out: mpsc::Sender<Event>,
+    ) -> Result<super::Source, String> {
+        kafka_source(self.clone(), out)
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+}
+
+fn kafka_source(
+    config: KafkaSourceConfig,
+    out: mpsc::Sender<Event>,
+) -> Result<super::Source, String> {
+    let consumer = Arc::new(create_consumer(config.clone())?);
+    let source = future::lazy(move || {
+        let hostname = hostname::get_hostname();
```

Comment:
Do you have a use case for including the hostname of the consumer? I know we do it for sources like syslog and tcp, but that makes a little more sense because we can get the address of the source machine. Here it's just our own host, which is easy enough to add later with an `add_fields` transform.

Reply:
Not really. I removed it for now to simplify the code.